qamomile/core/post_process/local_search.py

```diff
@@ -0,0 +1,108 @@
+"""
+qamomile/core/post_process.py
+
+This module provides functionality for Local Search Algorithm. Local Search Algorithm is one of the simplest 
+heuristic algorithms for the optimization problem. Let's take the Ising Hamiltonian as objective
+function to be minimized. 
+
+    .. math::
+        \sum_{ij} J_{ij} z_i z_j + \sum_i h_i z_i, ~\text{s.t.}~z_i \in \{-1, 1\}
+            
+Key Features:
+- Implementation of first- and best-improvement local search algorithms
+- Construction of cost Hamiltonians for the Ising model
+- Conversion of Ising interactions from dictionary format to matrix form
+"""
+
+from qamomile.core.ising_qubo import IsingModel
+from qamomile.core.converters.converter import QuantumConverter
+import qamomile.core.operator as qm_o
+from qamomile.core.converters.utils import is_close_zero
+from qamomile.core.bitssample import BitsSample, BitsSampleSet
+import numpy as np
+import jijmodeling as jm
+from typing import Callable
+
+class IsingMatrix:
+    def __init__(self, quad: np.array, linear: np.array):
+        self.quad = quad
+        self.linear = linear
+
+class LocalSearch:
+    def __init__(self, converter: QuantumConverter):
+        self.converter = converter
+        self.ising = converter.ising_encode()
+        
+    def decode(self, result) -> jm.experimental.SampleSet:
+        sample = BitsSample(1,np.where(result == -1, 0, 1).tolist())
+        sample_set = BitsSampleSet(bitarrays=[sample])
+        decoded_sampleset=self.converter.decode_bits_to_sampleset(sample_set)
+
+        return decoded_sampleset
+
+    def to_ising_matrix(self, ising: IsingModel) -> IsingMatrix:
+        size = max(max(i, j) for i, j in ising.quad.keys()) + 1
```

Comment:
`IsingModel` has `num_bits` to do this
https://github.com/Jij-Inc/Qamomile/blob/fb72a1325ed8cc42e759c4785bad9b48925f0b33/qamomile/core/ising_qubo.py#L12-L17

Reply:
You also used same code in the tutorial. Please fix it also.